espnet/bin/asr_train.py

```diff
@@ -156,12 +157,22 @@ def main(cmd_args):
     # minibatch related
     parser.add_argument('--sortagrad', default=0, type=int, nargs='?',
                         help="How many epochs to use sortagrad for. 0 = deactivated, -1 = all epochs")
-    parser.add_argument('--batch-size', '-b', default=50, type=int,
-                        help='Batch size')
+    parser.add_argument('--batch-count', default='auto', choices=BATCH_COUNT_CHOICES,
+                        help='How to count batch_size. The default (auto) will find how to count by args.')
+    parser.add_argument('--batch-size', '-b', default=0, type=int,
+                        help='Maximum seqs in a minibatch (0 to disable)')
+    parser.add_argument('--batch-bins', default=0, type=int,
+                        help='Maximum bins in a minibatch (0 to disable)')
+    parser.add_argument('--batch-frames-in', default=0, type=int,
+                        help='Maximum input frames in a minibatch (0 to disable)')
+    parser.add_argument('--batch-frames-out', default=0, type=int,
+                        help='Maximum output frames in a minibatch (0 to disable)')
+    parser.add_argument('--batch-frames-inout', default=0, type=int,
+                        help='Maximum input+output frames in a minibatch (0 to disable)')
     parser.add_argument('--maxlen-in', default=800, type=int, metavar='ML',
```

Comment:
We can unify the option names `--maxlen-in` --> `--batch-maxlen-in`, `--maxlen-out` --> `--batch-maxlen-out`
We can completely switch to `--batch-maxlen-xxx` and fix all `run.sh` or keep both `--batch-maxlen-xxx` and `--maxlen-xxx`

Reply:
I would like to keep both `--batch-frames-xxx` and  `--maxlen-xxx` because they play different roles.

Reply:
>     * it's better to move `batchfy.py` to `espnet/utils/batchfy.py` as it's used for both TTS and ASR.

agree

Reply:
I meant to say that we can just rename `--maxlen-xxx` to `--batch-maxlen-xxx` to follow the other naming conventions. You don't have to touch anything for `--maxlen-xxx`. Also, to clarify the role, we can rename them further as `--batch-seq-maxlen-xxx`. Again, by keeping the original `--maxlen-xxx` options, we'd not have confusions.

Reply:
I misunderstood. OK you mean
```python
parser.add_argument("--maxlen-in", "--batch-seq-maxlen-in", default=800, type=int, metavar='ML', ...)
```

Reply:
Correct.
With this, we can clarify the meaning of the option.